jfrog-cli/artifactory/utils/nuget/solution/solution_test.go

```diff
@@ -24,3 +30,128 @@ func TestEmptySolution(t *testing.T) {
+EndProject`, filepath.Join("jfrog", "path", "test", "packagesconfig.csproj"), "packagename"},
+		{"withpath", `Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "packagename", "packagesconfig/packagesconfig.csproj", "{D1FFA0DC-0ACC-4108-ADC1-2A71122C09AF}"
+EndProject`, filepath.Join("jfrog", "path", "test", "packagesconfig", "packagesconfig.csproj"), "packagename"},
+		{"sameprojectname", `Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "packagesconfig", "packagesconfig/packagesconfig.csproj", "{D1FFA0DC-0ACC-4108-ADC1-2A71122C09AF}"
+EndProject`, filepath.Join("jfrog", "path", "test", "packagesconfig", "packagesconfig.csproj"), "packagesconfig"},
+	}
+
+	path := filepath.Join("jfrog", "path", "test")
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			projectName, csprojPath, err := parseProject(test.projectLine, path)
+			if err != nil {
+				t.Error(err)
+			}
+			if csprojPath != test.expectedCsprojPath {
+				t.Error(fmt.Sprintf("Expected %s, got %s", test.expectedCsprojPath, csprojPath))
+			}
+			if projectName != test.expectedProjectName {
+				t.Error(fmt.Sprintf("Expected %s, got %s", test.expectedProjectName, projectName))
+			}
+		})
+	}
+}
+
+func TestGetProjectsFromSlns(t *testing.T) {
+	regExp, err := utils.GetRegExp(`Project\("(.*)\nEndProject`)
+	if err != nil {
+		t.Error(err)
+	}
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		t.Error(err)
+	}
+
+	testsDataDir := filepath.Join(pwd, "testsdata")
+	tests := []struct {
+		name             string
+		solution         solution
+		expectedProjects []string
+	}{
+		{"withoutSlnFile", solution{path: testsDataDir, slnFile: "", projects: nil}, []string{`Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "packagesconfigmulti", "packagesconfig.csproj", "{D1FFA0DC-0ACC-4108-ADC1-2A71122C09AF}"
+EndProject`, `Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "packagesconfiganothermulti", "test\packagesconfig.csproj", "{D1FFA0DC-0ACC-4108-ADC1-2A71122C09AF}"
+EndProject`, `Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "packagesconfig", "packagesconfig.csproj", "{D1FFA0DC-0ACC-4108-ADC1-2A71122C09AF}"
+EndProject`},
+		},
+		{"withSlnFile", solution{path: testsDataDir, slnFile: "oneproject.sln", projects: nil}, []string{`Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "packagesconfig", "packagesconfig.csproj", "{D1FFA0DC-0ACC-4108-ADC1-2A71122C09AF}"
+EndProject`},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			results, err := test.solution.getProjectsFromSlns(regExp)
+			if err != nil {
+				t.Error(err)
+			}
+			replaceCarriageSign(results)
+
+			if !reflect.DeepEqual(test.expectedProjects, results) {
+				t.Error(fmt.Sprintf("Expected %s, got %s", test.expectedProjects, results))
+			}
+		})
+	}
+}
+
+// Reading a file on windows adds the \r\n instead of just \n
+// This breaks string comparison.
+// Replaces each \r\n with just \n.
```

Comment:
The function comment may not be clear for any reader.
I suggest changing it to:
// If running on Windows, replace \r\n with \n.

Reply:
Done